### ISSUE TITLE
community.aws.aws_ssm not exiting after running command 

### ISSUE THREAD

**Issue by grvdeloitte** [ansible-collections/community.aws#1157]
### Summary

when i run command using ssm it is not exiting if commands run successfully. it is waiting for timeout.

### Issue Type

Bug Report

### Component Name

aws_ssm

<127.0.0.1> EXEC stdout line: {"changed": true, "stdout": "Linux xxxx 4.12.14-122.116-default #1 SMP Tue Apr 5 13:16:39 UTC 2022 (b42b08b) x86_64 x86_64 x86_64 GNU/Linux", "stderr": "", "rc": 0, "cmd": "uname -a", "start": "2022-05-25 01:09:30.376981", "end": "2022-05-25 01:09:30.381839", "delta": "0:00:00.004858", "msg": "", "invocation": {"module_args": {"_raw_params": "uname -a", "_uses_shell": true, "warn": false, "stdin_add_newline": true, "strip_empty_ends": true, "argv": null, "chdir": null, "executable": null, "creates": null, "removes": null, "stdin": null}}}
<127.0.0.1> EXEC remaining: 60
<127.0.0.1> EXEC remaining: 59
<127.0.0.1> EXEC remaining: 58
<127.0.0.1> EXEC remaining: 57
<127.0.0.1> EXEC remaining: 56
<127.0.0.1> EXEC remaining: 55
^C<127.0.0.1> CLOSING SSM CONNECTION TO: i-xxxx

### Ansible Version

```console (paste below)
$ ansible --version

```
ansible [core 2.11.9]
  config file = /etc/ansible/ansible.cfg
  configured module search path = ['/root/.ansible/plugins/modules', '/usr/share/ansible/plugins/modules']
  ansible python module location = /usr/local/lib/python3.6/site-packages/ansible
  ansible collection location = /etc/ansible/collections:/root/.ansible/collections:/usr/share/ansible/collections
  executable location = /bin/ansible
  python version = 3.6.8 (default, Sep 10 2021, 09:13:53) [GCC 8.5.0 20210514 (Red Hat 8.5.0-3)]
  jinja version = 2.10.1
  libyaml = True

### Collection Versions

```console (paste below)
$ ansible-galaxy collection list


# /etc/ansible/collections/ansible_collections
Collection    Version
------------- -------
amazon.aws    3.1.1
community.aws 3.1.0
steampunk.aws 0.8.4

# /root/.ansible/collections/ansible_collections
Collection    Version
------------- -------
amazon.aws    3.0.0
```

### AWS SDK versions

```console (paste below)
$ pip show boto boto3 botocore
```
Name: boto3
Version: 1.22.0
Summary: The AWS SDK for Python
Home-page: https://github.com/boto/boto3
Author: Amazon Web Services
Author-email: None
License: Apache License 2.0
Location: /root/.local/lib/python3.6/site-packages
Requires: s3transfer, jmespath, botocore
---
Name: botocore
Version: 1.25.0
Summary: Low-level, data-driven core of boto 3.
Home-page: https://github.com/boto/botocore
Author: Amazon Web Services
Author-email: None
License: Apache License 2.0
Location: /root/.local/lib/python3.6/site-packages
Requires: python-dateutil, urllib3, jmespath

### Configuration

```console (paste below)
$ ansible-config dump --only-changed

```

### Steps to Reproduce

```yaml (paste below)

```
- name: test connection
  hosts: localhost
  tasks:
    - name: Get AWS Access
      community.aws.sts_assume_role:
       role_arn: "arn:aws:iam::xxx:role/xxx"
       role_session_name: "testrole1"
      register: assumed_role

- name: test_file_name
  hosts: localhost
  gather_facts: false
  vars:
     ansible_aws_ssm_region: us-east-1
     ansible_aws_ssm_instance_id: "xxxxxx"
     ansible_connection: aws_ssm
     ansible_aws_ssm_bucket_name: xxxx
     ansible_aws_ssm_access_key_id: "{{ assumed_role.sts_creds.access_key }}"
     ansible_aws_ssm_secret_access_key: "{{ assumed_role.sts_creds.secret_key }}"
     ansible_aws_ssm_session_token: "{{ assumed_role.sts_creds.session_token }}"
  tasks:
      - name: Test file
        shell: uname -a

### Expected Results

i expected ssm to close connection when command ran succssfully.

### Actual Results

```console (paste below)

```
'
<127.0.0.1> EXEC stdout line: oFzeglPfjxKVluuJclMsdmGbJc
<127.0.0.1> EXEC stdout line:   % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
<127.0.0.1> EXEC stdout line:                                  Dload  Upload   Total   Spent    Left  Speed
100  124k  100  124k    0     0  1372k      0 --:--:-- --:--:-- --:--:-- 1372k
<127.0.0.1> EXEC stdout line:
<127.0.0.1> EXEC stdout line: 0
<127.0.0.1> EXEC stdout line: UvxqHPPWcmXBfzsgGnTLlcjBvo
<127.0.0.1> POST_PROCESS:   % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
100  124k  100  124k    0     0  1372k      0 --:--:-- --:--:-- --:--:-- 1372k

0
<127.0.0.1> (0, '  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current\r\r\n                                 Dload  Upload   Total   Spent    Left  Speed\r\r\n\r 0     0    0     0    0     0      0      0 --:--:-- --:--:-- --:--:--     0\r100  124k  100  124k    0     0  1372k      0 --:--:-- --:--:-- --:--:-- 1372k\r\r', '')
<127.0.0.1> (0, '  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current\r\r\n                                 Dload  Upload   Total   Spent    Left  Speed\r\r\n\r 0     0    0     0    0     0      0      0 --:--:-- --:--:-- --:--:--     0\r100  124k  100  124k    0     0  1372k      0 --:--:-- --:--:-- --:--:-- 1372k\r\r', '')
<127.0.0.1> EXEC chmod u+x /tmp//ansible/ansible-tmp-1653455368.1676185-211188-85325719141651/ /tmp//ansible/ansible-tmp-1653455368.1676185-211188-85325719141651/AnsiballZ_command.py
<127.0.0.1> _wrap_command: 'echo ndpgTebtZwFcXcwClvNKkeJynb
chmod u+x /tmp//ansible/ansible-tmp-1653455368.1676185-211188-85325719141651/ /tmp//ansible/ansible-tmp-1653455368.1676185-211188-85325719141651/AnsiballZ_command.py
echo $'\n'$?
echo zecjVeelXUiOEoBHdOlruYQarr
'
<127.0.0.1> EXEC stdout line: ndpgTebtZwFcXcwClvNKkeJynb
<127.0.0.1> EXEC stdout line:
<127.0.0.1> EXEC stdout line: 0
<127.0.0.1> EXEC stdout line: zecjVeelXUiOEoBHdOlruYQarr
<127.0.0.1> POST_PROCESS:
0
<127.0.0.1> (0, '\r', '')
<127.0.0.1> EXEC /usr/bin/python3.6 /tmp//ansible/ansible-tmp-1653455368.1676185-211188-85325719141651/AnsiballZ_command.py
<127.0.0.1> _wrap_command: 'echo uYQrEQDkifUMFwlFBTGbhhbbTo
sudo /usr/bin/python3.6 /tmp//ansible/ansible-tmp-1653455368.1676185-211188-85325719141651/AnsiballZ_command.py
echo $'\n'$?
echo GZSTOeaPGDIFXMHUVRpKWIGehb
'
<127.0.0.1> EXEC stdout line: uYQrEQDkifUMFwlFBTGbhhbbTo
<127.0.0.1> EXEC stdout line:
<127.0.0.1> EXEC stdout line: {"changed": true, "stdout": "Linux xxxx 4.12.14-122.116-default #1 SMP Tue Apr 5 13:16:39 UTC 2022 (b42b08b) x86_64 x86_64 x86_64 GNU/Linux", "stderr": "", "rc": 0, "cmd": "uname -a", "start": "2022-05-25 01:09:30.376981", "end": "2022-05-25 01:09:30.381839", "delta": "0:00:00.004858", "msg": "", "invocation": {"module_args": {"_raw_params": "uname -a", "_uses_shell": true, "warn": false, "stdin_add_newline": true, "strip_empty_ends": true, "argv": null, "chdir": null, "executable": null, "creates": null, "removes": null, "stdin": null}}}
<127.0.0.1> EXEC remaining: 60
<127.0.0.1> EXEC remaining: 59
<127.0.0.1> EXEC remaining: 58
<127.0.0.1> EXEC remaining: 57
<127.0.0.1> EXEC remaining: 56
<127.0.0.1> EXEC remaining: 55
^C<127.0.0.1> CLOSING SSM CONNECTION TO: i-xxx

### Code of Conduct

- [X] I agree to follow the Ansible Code of Conduct

**Comment by ansibullbot** [comment]
Files identified in the description:
* [`lib/ansible/plugins/connection`](https://github.com/['ansible-collections/amazon.aws', 'ansible-collections/community.aws', 'ansible-collections/community.vmware']/blob/main/lib/ansible/plugins/connection)
* [`plugins/connection/aws_ssm.py`](https://github.com/['ansible-collections/amazon.aws', 'ansible-collections/community.aws', 'ansible-collections/community.vmware']/blob/main/plugins/connection/aws_ssm.py)

If these files are inaccurate, please update the `component name` section of the description or use the `!component` bot command.

[click here for bot help](https://github.com/ansible/ansibullbot/blob/master/ISSUE_HELP.md)

**Comment by craigmiskell-gitlab** [comment]
We're seeing this too, on some but not all actions, in particular, a long-running apt-get install, where with a timeout of 120, it takes 60-70 seconds to execute, reports the entire output in "EXEC stdout" report, then continues counting down to 0 at which time it counts the task as succeeded.  Other short running tasks in the same ansible execution complete within 1-2 seconds and finish cleanly/quickly as expected.  

Given the while-loop at https://github.com/ansible-collections/community.aws/blob/main/plugins/connection/aws_ssm.py#L413 that *must* be returning None, which means the child process is still running so maybe the issue is more with SSM handling that particular command than with the code in this repo.

**Comment by tremble** [comment]
Thanks for taking the time to open this issue, and I'm sorry it's taken so long to get back to you.

With release 5.2.0 we've fixed a number of issues related to things getting "stuck".  Please could you update your collection and retry.

**Comment by alinabuzachis** [comment]
Closing this issue because I was not able to reproduce this issue with the latest release of the collection. However, feel free to open a new issue if the problem persists.
